modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/display/context/FDSViewsDisplayContext.java

```diff
@@ -41,6 +52,44 @@ public FDSViewsDisplayContext(
 
 		_portletRequest = portletRequest;
 		_serviceTrackerList = serviceTrackerList;
+
+		HttpServletRequest httpServletRequest =
+			PortalUtil.getHttpServletRequest(portletRequest);
+
+		_cetManager = (CETManager)httpServletRequest.getAttribute(
+			CETManager.class.getName());
+	}
+
+	public JSONArray getCellRendererCETJSONArray() {
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray();
+
+		List<FDSCellRendererCET> fdsCellRendererCETs = null;
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)_portletRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		try {
+			fdsCellRendererCETs = (List)_cetManager.getCETs(
+				themeDisplay.getCompanyId(), null,
+				ClientExtensionEntryConstants.TYPE_FDS_CELL_RENDERER,
+				Pagination.of(QueryUtil.ALL_POS, QueryUtil.ALL_POS), null);
+		}
+		catch (PortalException portalException) {
+			_log.error(
+				"Unable to get FDS Cell Renderer client extension entries",
+				portalException);
+		}
+
+		for (FDSCellRendererCET fdsCellRenderer : fdsCellRendererCETs) {
```

Comment:
When I add `return Collections.emptyList();` to the `catch (PortalException portalException) {` I get the following error:

```bash
Task :apps:frontend-data-set:frontend-data-set-views-web:compileJava FAILED
/Users/kresimircoko/Documents/Work/Projects/dxp/liferay-portal/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/display/context/FDSViewsDisplayContext.java:78: error: incompatible types: no instance(s) of type variable(s) T exist so that java.util.List<T> conforms to com.liferay.portal.kernel.json.JSONArray
                        return Collections.emptyList();
                                                    ^
```

I have no idea what it means + VSC isn't offering me any tips on how to handle it or underlining errors so 🤷

Reply:
Ah, sorry, my bad. That method is returning a `JSONArray` and not a `List`. I was looking at `List<FDSCellRendererCET> fdsCellRendererCETs`.

So, you need to return an empty `JSONArray`:
```java
return JSONFactoryUtil.createJSONArray();
```

or `return jsonArray` at this point, but I think explicitly returning an empty is slightly better practice. Later on, somebody might add code that modifies `jsonArray` before the `try` block.